### ISSUE TITLE
Модуль голосовой активации на основе cmu sphinx

### ISSUE THREAD

**Issue by ivanbalashov239** [uzyovoys/aggregate#2]
http://cmusphinx.sourceforge.net/
просьба проголосовать на этом сайте за создание модели русского языка. С ней будет возможность использовать сфинкс как модуль распознавания речи, но и сейчас его можно использовать для активации, натренировав на одно единственное слово/фразу, если я не ошибаюсь это доступно.

**Comment by morfeusys** [comment]
Конечно доступно. Такой модуль можно создать уже сейчас. Для этого можно использовать pocketsphinx. Активация голосом будет примерно как в Дусе. Модель русского языка уже давно есть.

**Comment by ivanbalashov239** [comment]
@Morfeusys действительно есть, опрос сбил с толку, https://sourceforge.net/projects/cmusphinx/files/Acoustic%20and%20Language%20Models/Russian/

 будет время, попробую сделать, если никто не опередит.

**Comment by ivanbalashov239** [comment]
@Morfeusys нашел инфу, что вы уже работали с pocketsphinx , [вот например](https://github.com/Morfeusys/veravoice)  сюда не приделаете? Напишите где нибудь, запланированные к реализации функции, что бы суеты(моей в том числе) не было.

**Comment by pastukhov** [comment]
:+1:

**Comment by morfeusys** [comment]
Нет, я сам еще ничего не планирую. Если вы будете реализовывать - отлично. Но в агрегате в самом ядре кое-что нужно еще доделать для этого. Так как голосовая активация активировать-то может, а вот сказать, когда ЗАКАНЧИВАТЬ слушать - нет.

**Comment by ivanbalashov239** [comment]
@Morfeusys  ясно, тогда повременим

**Comment by morfeusys** [comment]
@seniorivn саму активацию разработать можно пока - ничего не мешает, Просто в виде теста. А агрегат до новой версии я обновлю довольно быстро.

**Comment by pastukhov** [comment]
Заканчивать по таймауту и по детекторы тишины. Sox это умеет делать.

**Comment by ivanbalashov239** [comment]
@pastukhov в #36  я уже привел этот вариант, но @Morfeusys сказал, что сделает сам, используя алгоритм гейта(пока не знаком)

**Comment by morfeusys** [comment]
@pastukhov @seniorivn sox как раз и реализует такой алгоритм

**Comment by morfeusys** [comment]
@seniorivn @pastukhov Кажется, я нашел кое-что интереснее для случаев, когда доступны иксы или на windows ;-)

**Comment by ivanbalashov239** [comment]
@Morfeusys  а маки в пролете? ссылку/пояснение дашь? или сам просто сделаешь?
П.С. надо бы сделать todo list

**Comment by morfeusys** [comment]
@seniorivn почему в пролете? Нет, все в силе )) неправильно выразился. Я сейчас реализую как раз, займет время. Используется Web Speech API 

todolist лучше всего реализовать в каком-то отдельном сервисе коллаборации, либо здесь навешивать лейблы на issue, что я и делаю.

**Comment by ivanbalashov239** [comment]
@Morfeusys лейблы видел, это хорошо, но todolist всё равно нужен, какой не знаю, в крайнем случае здесь есть свои:

``` markdown
- [] for an empty checkbox
- [x] for a checked checkbox
```
- [ ] for an empty checkbox
- [x] for a checked checkbox

**Comment by morfeusys** [comment]
В 0.0.2 теперь есть событие asr.listen - с его использованием можно смело делать голосовоую активацию!

**Comment by PaulAnnekov** [comment]
Я уже начал делать на pocketsphinx + ruby swig. Ruby враппер выбран потому что он более-менее кросплатформенный и vert.x поддерживает этот язык. Делаю медленно ибо в свободное от работы время...
